EPMLSTRTAI-338: More detailed exception

Walrus94 opened the pull request.

iteaky committed:
EPMLSTRTAI-309: Remove getExceptionMessage from Validator interface.
 Add getExceptionMessage to AbstractValidator with SearchRule and ValidationContext as new parameters.
 Add ValidationContext to ValidationResult.
 Override getExceptionMessage for UniquenessLocatorValidator.

pomkine commented on src/main/java/com/epam/page/object/generator/validators/UniquenessLocatorValidator.java:
```diff
@@ -40,7 +43,21 @@ public boolean isValid(SearchRule searchRule, ValidationContext validationContex
+                    return null;
+                })
+                .filter(pair -> pair.getValue() > 1)
+                .map(pair -> "On url: " + pair.getKey() + " founded " + pair.getValue() + " elements")
```
founded? or found? :)

pomkine commented on src/main/java/com/epam/page/object/generator/validators/ValidationResult.java:
```diff
@@ -5,15 +5,17 @@
 
     private boolean isValid;
-    private Validator validator;
+    private AbstractValidator validator;
```
use interface instead

kugushev commented on src/main/java/com/epam/page/object/generator/validators/UniquenessLocatorValidator.java:
```diff
@@ -40,7 +43,21 @@ public boolean isValid(SearchRule searchRule, ValidationContext validationContex
+        return validationContext.getUrls()
+                .stream()
+                .map(url -> {
+                    try {
```
7-lines lambda is a bad style. Let's extract a method

iteaky replied:
@AleksandrKugushev  just extract lambda to method and use it inside method getExceptionMessage()???
or exctract some part of lambda?

iteaky committed:
EPMLSTRTAI-309:Remove validator from validationResult, fix tests

iteaky committed:
EPMLSTRTAI-309:small fixes

iteaky committed:
EPMLSTRTAI-309:improvements with comments on github:
1) rename founded -> found
2) Extract method with try-catch from lambda in getException()
3) No more exist of validator in ValidationResult

iteaky committed:
EPMLSTRTAI-309:
uniqueness -> unique

pomkine reviewed (changes requested):
Don't forget to fix new codeclimate issues (except imports). It is mainly about a length of lines (shoulb be < 100)

iteaky committed:
EPMLSTRTAI-338 code climate fixes

iteaky committed:
EPMLSTRTAI-338 code climate fixes

iteaky committed:
EPMLSTRTAI-338 code climate fixes
Wrong lexicographical order for 'java.util.stream.Collectors' import. Should be before 'javafx.util.Pair'.
